File: ecommerce_app/lib/data/models/product_model.dart
class ProductModel {
  String? sId;
  String? category;
  String? title;
  String? description;
  int? price;
  List<String>? images;
  String? updatedAt;
  String? createdAt;

  ProductModel(
      {this.sId,
      this.category,
      this.title,
      this.description,
      this.price,
      this.images,
      this.updatedAt,
      this.createdAt});

  ProductModel.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    category = json['category'];
    title = json['title'];
    description = json['description'];
    price = json['price'];
    images = json['images'].cast<String>();
    updatedAt = json['updatedAt'];
    createdAt = json['createdAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.sId;
    data['category'] = this.category;
    data['title'] = this.title;
    data['description'] = this.description;
    data['price'] = this.price;
    data['images'] = this.images;
    data['updatedAt'] = this.updatedAt;
    data['createdAt'] = this.createdAt;
    return data;
  }
}

File: ecommerce-backend/src/models/product_model.js
const { Schema, model } = require("mongoose");

const productSchema = new Schema({
	category: { type: Schema.Types.ObjectId, ref: "Category", required: true },
	title: { type: String, required: [true, "title is required"] },
	description: { type: String, default: "" },
	price: { type: Number, required: true },
	images: { type: Array, default: [] },
	updatedAt: { type: Date },
	createdAt: { type: Date },
});

productSchema.pre("save", function (next) {
	this.updatedAt = new Date();
	this.createdAt = new Date();

	next();
});

productSchema.pre(["update", "findOneAndUpdate", "updateOne"], function (next) {
	const update = this.getUpdate();
	delete update._id;

	this.updatedAt = new Date();

	next();
});

const ProductModel = model("Product", productSchema);

module.exports = ProductModel;
